overlord/devicestate/devicemgr.go

```diff
@@ -556,6 +556,22 @@ func (m *DeviceManager) ensureOperational() error {
 		}
 		hasPrepareDeviceHook = (gadgetInfo.Hooks["prepare-device"] != nil)
 	}
+	if device.KeyID == "" && model.Grade() != "" {
+		// UC20+ devices support factory reset
+		keyID, serial, err := m.maybeRestoreAfterReset(device)
+		if err != nil {
+			return err
+		}
+		if keyID != "" && serial != "" {
+			logger.Debugf("restored device key %v and serial %v", keyID, serial)
```

Comment:
this probably should be a Noticef, no?

Reply:
Also, shouldn't this message go below `m.setDevice(..)` (so it's not logged in case of error)?

Reply:
moved it below